### PR TITLE
Added lookuptable testcases

### PR DESCRIPTION
**Pull request by sameena1415**
## Summary
Automated 15 test cases for lookup table functionality. 
Test plan link: https://docs.google.com/spreadsheets/d/1-Fqax67kLz8dIJE5UcM1iGeWCV9mAm5OH6lZwb290cQ/edit#gid=0

## Description
Automated 15 test cases for lookup table functionality. 
Test plan link: https://docs.google.com/spreadsheets/d/1-Fqax67kLz8dIJE5UcM1iGeWCV9mAm5OH6lZwb290cQ/edit#gid=0

### Link to Jira Ticket (if applicable)
[Jira ticket link ](https://dimagi-dev.atlassian.net/browse/QA-4402)

## QA Checklist

- [X] Label(s) and Assignee added
- [X] PR sent for review
- [ ] Documentation (if applicable) added/updated

### CODE REVIEW

File: common_utilities/Excel/excel_manage.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        filepath = self.path
+        wb.save(filepath)
+
+    def create_sheet(self, sheet_name):
+        wb = load_workbook(self.path)
+        #sheet = wb.active
+        #sheet.title = sheet_name
+        #wb.create_sheet(index=1, title=sheet_name)
+        #wb.create_sheet(title=sheet_name)
+        wb.create_sheet(sheet_name)
```
Comment (kbo001) [inline common_utilities/Excel/excel_manage.py:24]: you can remove the commented codes if not needed for future.

File: Lookuptable/testPages/data/export_data_page.py [hunk 2]
```diff
@@ -0,0 +1,489 @@
+import os
+import time
+import pandas as pd
+from datetime import datetime, timedelta
+
+from common_utilities.selenium.base_page import BasePage
+from common_utilities.path_settings import PathSettings
+from HQSmokeTests.userInputs.user_inputs import UserData
```
Comment (kbo001) [inline Lookuptable/testPages/data/export_data_page.py:8]: rename the reference

File: Lookuptable/testPages/data/export_data_page.py [hunk 3]
```diff
@@ -0,0 +1,489 @@
+import os
```
Comment (kbo001) [inline Lookuptable/testPages/data/export_data_page.py:1]: Also refer to this class in your testcases instead of the HQSmokeTest one

**Review by kbo001 — CHANGES_REQUESTED** [review]
@sameena1415 Please make the changes requested

File: Lookuptable/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+## Commcare lookuptable Test Script
+
+Smoke tests ensure that the important features are working and there are no showstoppers in the build deployed to environments.\
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/README.md:3]: Done. Thank you !!

File: Lookuptable/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+- You could also pass the following arguments
+  - ` -n 3 --dist=loadfile` - This will run the tests parallelly in 3 instances. The number of reruns is configurable.
+  - ` --reruns 1` - This will re-run the tests once in case of failures.The number of reruns is configurable too.
+
+### <ins> Trigger Manually on Gitaction </ins>
+
+<img align="right" width="400" src="https://user-images.githubusercontent.com/67914792/168757107-3ce9bb6a-57b5-4c15-b20d-e7883bf9ed65.PNG" alt="clone this repository" />
+
+To manually trigger the script,
+  - Go to [HQ Smoke Tests action](https://github.com/dimagi/dimagi-qa/actions/workflows/python-app.yml)
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/README.md]: Changes made and pushed the new changes . Thank you !!

File: Lookuptable/email_fail.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+Hello!
+
+The CommCare HQ Smoke Test run has FAILED on "{{environment}}"! Please check the attachment to see the results.
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/email_fail.md]: Changes made and pushed the new changes . Thank you !!

File: Lookuptable/email_pass.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+Hello!
+
+All the CommCare HQ Smoke Tests PASSED successfully on "{{environment}}".
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/email_pass.md]: Changes made and pushed the new changes . Thank you !!

File: Lookuptable/requires.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+pytest-html>=3.1.1
+selenium>=3.141.0
+webdriver-manager>=3.2.2
+openpyxl>=3.0.6
+matplotlib >= 3.3.4
+pytest-rerunfailures>=10.2
+pytest-xdist>=2.4.0
+pytest-xdist[psutil]
+Appium-Python-Client == 0.52
+pyotp >=2.6.0
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/requires.txt:14]: Changes made and pushed the new changes . Thank you !!

File: Lookuptable/settings-sample.cfg [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+url = https://www.commcarehq.org/
+# Login username of the webuser
+login_username =
+# Login password of the webuser
+login_password =
+# This is a preconfigured browsertack username for mobile tests
+bs_user =
+# This is a preconfigured browsertack key used for mobile tests
+bs_key =
+# This is a preconfigured authentication key used for 2FA tests on staging
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/settings-sample.cfg]: Changes made and pushed the new changes . Thank you !!

File: Lookuptable/settings-sample.cfg [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+staging_auth_key =
+# This is a preconfigured authentication key used for 2FA tests on prod
+prod_auth_key =
+# This is a preconfigured password for the user used to test web user invitation tests
+invited_webuser_password =
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/settings-sample.cfg]: Changes made and pushed the new changes . Thank you !!

File: Lookuptable/testCases/conftest.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+                DIMAGIQA_MAIL_USERNAME
+                DIMAGIQA_MAIL_PASSWORD
+
+            See https://docs.github.com/en/actions/reference/encrypted-secrets
+            for instructions on how to set them.
+            """
+    settings = {}
+    for name in ["url", "login_username", "login_password", "mail_username",
+                 "mail_password", "bs_user", "bs_key", "staging_auth_key", "prod_auth_key", "invited_webuser_password"]:
+
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/conftest.py:29]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/conftest.py [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+@pytest.fixture(scope="session", autouse=True)
+def settings(environment_settings_hq):
+    if os.environ.get("CI") == "true":
+        settings = environment_settings_hq
+        settings["CI"] = "true"
+        if any(x not in settings for x in ["url", "login_username", "login_password",
+                                           "mail_username", "mail_password", "bs_user", "bs_key", "staging_auth_key",
+                                           "prod_auth_key", "invited_webuser_password"]):
+            lines = environment_settings_hq.__doc__.splitlines()
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/conftest.py:50]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_download_update.py [hunk 13]
```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+
+import pytest
+
+from HQSmokeTests.testPages.data.export_data_page import ExportDataPage
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/test_download_update.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_download_update.py [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+@pytest.mark.excel
+def test_08_download_update_4(driver):
+    data = LookUpTablePage(driver)
+    export = ExportDataPage(driver)
+    export.data_tab()
+    values['table_id'] = data.create_download_lookuptable()
+    excel = ExcelManager(driver)
+    Download_path = data.error_upload1()
+    excel.__init__(Download_path)
+    excel = ExcelManager(Download_path)
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/test_download_update.py:63]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_lookuptable_creation.py [hunk 15]
```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+
+import pytest
+
+from HQSmokeTests.testPages.data.export_data_page import ExportDataPage
+from common_utilities.Excel.excel_manage import ExcelManager
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/test_lookuptable_creation.py:6]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_lookuptable_creation.py [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    Download_path = data.error_upload1()
+    print("path is " + Download_path)
+    excel = ExcelManager(Download_path)
+    for x in UserData.Col_headers:
+        for i in range(1,2):
+            col = excel.col_size(values['table_id'])
+            excel.write_excel_data(values['table_id'],1,col+i,x)
+    for y in range(1,3):
+            excel.write_data(values['table_id'], UserData.data_list1)
+    data = LookUpTablePage(driver)
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/test_lookuptable_creation.py:77]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_upload.py [hunk 17]
```diff
@@ -0,0 +1,70 @@
+from pathlib import Path
+
+import pytest
+
+from HQSmokeTests.testPages.data.export_data_page import ExportDataPage
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/test_upload.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/applications/app_preview.py [hunk 18]
```diff
@@ -0,0 +1,51 @@
+import time
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/applications/app_preview.py:1]: We must incorporate the fixture at the application level and submit the form for the further  test cases.

File: Lookuptable/testPages/applications/application_page.py [hunk 19]
```diff
@@ -0,0 +1,217 @@
+import time
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/applications/application_page.py:1]: We must incorporate the fixture at the application level and submit the form for the further test cases.

File: Lookuptable/testPages/data/lookup_table_page.py [hunk 20]
```diff
@@ -0,0 +1,199 @@
+import time
+import glob
+import os.path
+from pathlib import Path
+
+
+from selenium.webdriver.common.by import By
+from HQSmokeTests.testPages.data.export_data_page import ExportDataPage
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/data/lookup_table_page.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/data/lookup_table_page.py [hunk 21]
```diff
@@ -0,0 +1,199 @@
...
+""""Contains test page elements and functions related to the Lookup Table module"""
+
+
+class LookUpTablePage(BasePage):
+
+    def __init__(self, driver):
+        super().__init__(driver)
+
+        self.table_id_name = "lookuptable_" + str(fetch_random_string())
+        self.dummyid = "!@@#%%^^ Lookup"
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/data/lookup_table_page.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/data/lookup_table_page.py [hunk 22]
```diff
@@ -0,0 +1,199 @@
...
+    def invalid_data_assert(self):
+        invalid_data = self.get_text(self.erroralert_msg)
+        print(invalid_data)
+        assert invalid_data == UserData.invalid_data_assert
+
+    def missing_data_assert(self):
+        missing_data = self.get_text(self.errorUploadmsg)
+        new = missing_data.split(':')[0]
+        print("new+++",new,"***")
+        print("123",UserData.missing_data_assert,"###")
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/data/lookup_table_page.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/data/lookup_table_page.py [hunk 23]
```diff
@@ -0,0 +1,199 @@
...
+    def download1(self):
+        self.wait_to_click(self.manage_tables_link)
+        self.wait_to_click(self.select_checkbox)
+        print(self.select_checkbox)
+        self.wait_to_click(self.click_download)
+        self.wait_to_click(self.download_file)
+        time.sleep(3)
+        self.wait_to_click(self.closedownloadpopup)
+
+    def download1SpecificTable(self):
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/data/lookup_table_page.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/home/home_page.py [hunk 24]
```diff
@@ -0,0 +1,82 @@
+from selenium.webdriver.common.by import By
+
+from common_utilities.selenium.base_page import BasePage
+from HQSmokeTests.userInputs.user_inputs import UserData
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testPages/home/home_page.py]: Changes made and pushed the new changes. Thank you !!

File: common_utilities/Excel/excel_manage.py (hunk 1 above)
Comment (sameena1415) [reply common_utilities/Excel/excel_manage.py:25]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/data/export_data_page.py (hunk 2 above)
Comment (sameena1415) [reply Lookuptable/testPages/data/export_data_page.py:8]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testPages/data/export_data_page.py (hunk 3 above)
Comment (sameena1415) [reply Lookuptable/testPages/data/export_data_page.py:1]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_lookuptable_creation.py [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+@pytest.mark.lookupexcel
+def test_21_Error_upload3(driver):
+    export = ExportDataPage(driver)
+    export.data_tab()
+    data = LookUpTablePage(driver)
+    values['table_id'] = data.create_download_lookuptable()
+    Download_path = data.error_upload1()
+    print("path is " + Download_path)
+    data.write_data_excel(values['table_id'], Download_path)
+    data = LookUpTablePage(driver)
```
Comment (kbo001) [inline Lookuptable/testCases/test_lookuptable_creation.py:77]: You have already created an object of LookUpTablePage in line 72. You can remove this one. One instance is enough per test.

File: Lookuptable/testCases/test_upload.py [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+    export.data_tab()
+    data = LookUpTablePage(driver)
+    values['table_id']=data.create_download_lookuptable()
+    download_path = data.error_upload1()
+    print("path is " + download_path)
+    excel = ExcelManager(download_path)
+    col = excel.col_size(values['table_id'])
+    print("table_id", values['table_id'])
+    excel.write_excel_data(values['table_id'], 1, col + 1, "user 1")
+    excel.upload_to_path(values['table_id'], UserData.data_list)
```
Comment (kbo001) [inline Lookuptable/testCases/test_upload.py:36]: Similarly, create a function for steps from 32 to 36. If the steps are similar to `write_data_excel()` then use that function here as well. Make sure to optimize the testcases.

File: Lookuptable/testCases/test_upload.py [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+    export.data_tab()
+    data = LookUpTablePage(driver)
+    values['table_id']=data.create_download_lookuptable()
+    download_path = data.error_upload1()
+    print("path is " + download_path)
+    excel = ExcelManager(download_path)
+    col = excel.col_size(values['table_id'])
+    print("table_id", values['table_id'])
+    excel.write_excel_data(values['table_id'], 1, col + 1, "group 1")
+    excel.upload_to_path(values['table_id'], UserData.data_list)
```
Comment (kbo001) [inline Lookuptable/testCases/test_upload.py:52]: Same goes for steps 48 to 52

File: Lookuptable/userInputs/user_inputs.py [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+    type_data_list = [('N', fieldval, 'yes',fieldval)]
+    typeSheet_Headers = [('UID', 'Delete(Y/N)', 'field: '+ fieldval)]
+    data_upload_path = PathSettings.ROOT + "\\upload_1.xlsx"
+    print(data_upload_path)
+    hypertension_upload_path = PathSettings.ROOT + "\\Hypertension.xlsx"
+    MalformedDocument_upload_path = PathSettings.ROOT + "\\MalformedDocument.xlsx"
+    invalid_data_assert = "However, we ran into the following problems:"
+    missing_data_assert = "Please fix the following formatting issues in your Excel file"
+    Col_headers = ["user 1", "group 1"]
+
```
Comment (kbo001) [inline Lookuptable/userInputs/user_inputs.py:26]: variable names should start with lowercase. Also, words like `MalformedDocument_upload_path` should have been `malformed_document_upload_path`. This is just an example. Going forward keep this practice.

File: common_utilities/Excel/excel_manage.py [hunk 29]
```diff
@@ -0,0 +1,127 @@
+import os
```
Comment (kbo001) [inline common_utilities/Excel/excel_manage.py:1]: why is os imported? Is it used anywhere in this class?

File: common_utilities/Excel/excel_manage.py (hunk 1 above)
Comment (kbo001) [reply common_utilities/Excel/excel_manage.py:25]: I do not see the changes. I meant that you remove the commented code if they are not required. To clean the code.

**Review by kbo001 — CHANGES_REQUESTED** [review]
Please make the changes requested @sameena1415

**Comment by kbo001** [comment]
@sameena1415 create the worflow yml file for the lookup table scripts so that we can check if the code will get properly executed in git

File: Lookuptable/README.md [hunk 30]
```diff
@@ -0,0 +1,57 @@
...
+- You could also pass the following arguments
+  - ` -n 3 --dist=loadfile` - This will run the tests parallelly in 3 instances. The number of reruns is configurable.
+  - ` --reruns 1` - This will re-run the tests once in case of failures.The number of reruns is configurable too.
+
+### <ins> Trigger Manually on Gitaction </ins>
+
+<img align="right" width="400" src="https://user-images.githubusercontent.com/67914792/168757107-3ce9bb6a-57b5-4c15-b20d-e7883bf9ed65.PNG" alt="clone this repository" />
+
+To manually trigger the script,
+  - Go to [Lookuptable action](https://github.com/sameena1415/dimagi-qa/actions/lookuptable.yml)
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/README.md]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/settings-sample.cfg [hunk 31]
```diff
@@ -0,0 +1,10 @@
+[default]
+# This is the environment url of commcare
+url = https://www.commcarehq.org/
+# Login username of the webuser
+login_username =
+# Login password of the webuser
+login_password =
+# This is a preconfigured browsertack username for mobile tests
+staging_auth_key =
+# This is a preconfigured authentication key used for 2FA tests on prod
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/settings-sample.cfg]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_download_update.py [hunk 32]
```diff
@@ -0,0 +1,67 @@
...
+
+@pytest.mark.data
+@pytest.mark.managetables
+@pytest.mark.excel
+def test_07_download_update_3(driver):
+    export = ExportDataPage(driver)
+    export.data_tab()
+    data = LookUpTablePage(driver)
+    values['table_id'] = data.create_download_lookuptable()
+    Download_path = data.error_upload1()
```
[the remark this answers is not in the document]
Comment (sameena1415) [reply Lookuptable/testCases/test_download_update.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/testCases/test_upload.py (hunk 27 above)
Comment (sameena1415) [reply Lookuptable/testCases/test_upload.py]: Changes made and pushed the new changes. Thank you !!

File: Lookuptable/userInputs/user_inputs.py (hunk 28 above)
Comment (sameena1415) [reply Lookuptable/userInputs/user_inputs.py:26]: Changes made and pushed the new changes. Thank you !!

File: common_utilities/Excel/excel_manage.py (hunk 29 above)
Comment (sameena1415) [reply common_utilities/Excel/excel_manage.py]: Changes made and pushed the new changes. Thank you !!

**Review by kbo001 — COMMENTED** [review]
@kiran-tripathy when you get time, please review the code that Sameena pushed.

File: Lookuptable/README.md [hunk 33]
```diff
@@ -0,0 +1,57 @@
+## Commcare lookuptable Test Script
+
+Lookup table tests ensure that the important features are working and there are no showstoppers in the build deployed to environments.\
+The automated tests comprises of [these Lookuptable functionality.](https://docs.google.com/spreadsheets/d/1-Fqax67kLz8dIJE5UcM1iGeWCV9mAm5OH6lZwb290cQ/edit#gid=0)
```
Comment (kiran-tripathy) [inline Lookuptable/README.md:4]: Can we please link the master lookup table test plan here instead: https://docs.google.com/spreadsheets/d/1aTBv4zGeVuNTdnCyuleZbbpqOhPEGl0f78rcKAGWQHQ/edit#gid=1360466241?

File: Lookuptable/README.md [hunk 34]
```diff
@@ -0,0 +1,57 @@
+## Commcare lookuptable Test Script
+
+Lookup table tests ensure that the important features are working and there are no showstoppers in the build deployed to environments.\
```
Comment (kiran-tripathy) [inline Lookuptable/README.md:3]: Let's update the description here to:

`These tests ensure that the lookup table features work as expected and that there are no regressions.`

File: Lookuptable/requires.txt [hunk 35]
```diff
@@ -0,0 +1,15 @@
+## Stores information about all the libraries, modules, and packages that are used in this project.
+
+flake8>=3.8.4
```
Comment (kiran-tripathy) [inline Lookuptable/requires.txt:3]: Are we using the library `flake8` and `matplotlib` here?

File: Lookuptable/settings-sample.cfg [hunk 36]
```diff
@@ -0,0 +1,12 @@
...
+prod_auth_key =
```
Comment (kiran-tripathy) [inline Lookuptable/settings-sample.cfg:11]: Are we making use of these?
Also, the description says `# This is a preconfigured BrowserStack username for mobile tests`. I don't think we're doing any mobile-related tests here.

File: Lookuptable/testCases/test_download_update.py [hunk 37]
```diff
@@ -0,0 +1,69 @@
...
+@pytest.mark.managetables
+@pytest.mark.excel
+def test_07_download_update_3(driver):
+    export = ExportDataPage(driver)
+    export.data_tab()
+    data = LookUpTablePage(driver)
+    values['table_id'] = data.create_download_lookuptable()
+    download_path = data.error_upload1()
+    excel = ExcelManager(download_path)
+    print("table_id", values['table_id'])
```
Comment (kiran-tripathy) [inline Lookuptable/testCases/test_download_update.py:44]: Are these print statements required?

File: Lookuptable/testPages/data/export_data_page.py (hunk 2 above)
Comment (kiran-tripathy) [reply Lookuptable/testPages/data/export_data_page.py:8]: I still see the reference to `HQSmokeTests`. Am I missing something here?

File: Lookuptable/testPages/applications/application_page.py [hunk 38]
```diff
@@ -0,0 +1,217 @@
...
+from common_utilities.selenium.base_page import BasePage
+from common_utilities.generate_random_string import fetch_random_string
+from HQSmokeTests.userInputs.user_inputs import UserData
+from common_utilities.path_settings import PathSettings
+from HQSmokeTests.testPages.users.org_structure_page import latest_download_file
+
+""""Contains test page elements and functions related to the applications on Commcare"""
+
+
+class ApplicationPage(BasePage):
```
Comment (kiran-tripathy) [inline Lookuptable/testPages/applications/application_page.py:15]: I don't think there's any need to add redundant classes when we have existing classes here https://github.com/dimagi/dimagi-qa/tree/master/HQSmokeTests/testPages/applications

we can simply create objects for the existing classes to call the required functions.

File: Lookuptable/testPages/data/export_data_page.py [hunk 39]
```diff
@@ -0,0 +1,489 @@
...
+
+def latest_download_file():
+    os.chdir(PathSettings.DOWNLOAD_PATH)
+    files = sorted(os.listdir(os.getcwd()), key=os.path.getmtime)
+    newest = max(files, key=os.path.getctime)
+    print("File downloaded: " + newest)
+    return newest
+
+
+class ExportDataPage(BasePage):
```
Comment (kiran-tripathy) [inline Lookuptable/testPages/data/export_data_page.py:25]: Same! Redundant code. We can reference https://github.com/dimagi/dimagi-qa/blob/master/HQSmokeTests/testPages/data/export_data_page.py

File: Lookuptable/testPages/home/home_page.py [hunk 40]
```diff
@@ -0,0 +1,82 @@
+from selenium.webdriver.common.by import By
+
+from common_utilities.selenium.base_page import BasePage
+from Lookuptable.userInputs.user_inputs import UserData
+
+""""Contains test page elements and functions related to the Homepage of Commcare"""
+
+
+class HomePage(BasePage):
```
Comment (kiran-tripathy) [inline Lookuptable/testPages/home/home_page.py:9]: Again Redundant! [This ](https://github.com/dimagi/dimagi-qa/blob/master/HQSmokeTests/testPages/home/home_page.py) should be referenced instead, wherever its used.

**Review by kiran-tripathy — CHANGES_REQUESTED** [review]
@sameena1415 Firstly, great work here! 
I've added some comments; please go through those and make the necessary changes.
I'd encourage you to create an individual commit for each change so that you can link the commit id before marking any specific comment as resolved.

Lastly, could we please move all the `.xlsx` files to one directory, maybe somewhere in `userInputs`?

Let me or Kankana know if any question arises.

File: Lookuptable/README.md (hunk 34 above)
Comment (kiran-tripathy) [reply Lookuptable/README.md:3]: Let's update the description here to:

```suggestion
These tests ensure that the [lookup table ](https://confluence.dimagi.com/display/commcarepublic/Lookup+Tables)features work as expected and that there are no regressions.

```

File: Lookuptable/README.md (hunk 33 above)
Comment (sameena1415) [reply Lookuptable/README.md]: changes made . Here is the commit id for reference: 99b00a9f008a562cec8a5a54baac05797b07c2c2

File: Lookuptable/requires.txt (hunk 35 above)
Comment (sameena1415) [reply Lookuptable/requires.txt]: We are not using it. So I have updated and pushed the code. Commit id : d09e3f95d7bc85be69e2ee9de0612cc5851ba023

File: Lookuptable/settings-sample.cfg (hunk 36 above)
Comment (sameena1415) [reply Lookuptable/settings-sample.cfg]: Changes done. Commit id: 0d84b418e608b3130b5c249c9d1f9a93b7b7c14f

File: Lookuptable/testPages/data/export_data_page.py (hunk 39 above)
Comment (sameena1415) [reply Lookuptable/testPages/data/export_data_page.py]: Changes made. Commit code for reference : fccca641344bce18df6c10d99660adae9e687091

File: Lookuptable/testPages/home/home_page.py (hunk 40 above)
Comment (sameena1415) [reply Lookuptable/testPages/home/home_page.py]: Changes made. Commit code for reference : fccca641344bce18df6c10d99660adae9e687091

**Comment by sameena1415** [comment]
> @sameena1415 Firstly, great work here! I've added some comments; please go through those and make the necessary changes. I'd encourage you to create an individual commit for each change so that you can link the commit id before marking any specific comment as resolved.
> 
> Lastly, could we please move all the `.xlsx` files to one directory, maybe somewhere in `user inputs`?
> 
> Let me or Kankana know if any question arises.


@kiran-tripathy  I have made the changes and here is the commit code for reference: fccca641344bce18df6c10d99660adae9e687091

File: Lookuptable/testCases/test_download_update.py (hunk 37 above)
Comment (sameena1415) [reply Lookuptable/testCases/test_download_update.py]: Changes made. Commit code for reference: fccca641344bce18df6c10d99660adae9e687091

File: Features/Lookuptable/README.md [hunk 41]
```diff
@@ -0,0 +1,57 @@
...
+
+
+ -   Copy `settings-sample.cfg` to `settings.cfg` and populate `settings.cfg` for
+the environment you want to test.
+- Run tests using pytest command like:
+
+```sh
+
+# To execute all the test cases 
+pytest -v --rootdir= Lookuptable/testCases
```
Comment (kiran-tripathy) [inline Features/Lookuptable/README.md:36]: path needs to be updated @sameena1415

Comment (kiran-tripathy) [reply Features/Lookuptable/README.md:36]: https://github.com/dimagi/dimagi-qa/pull/169/commits/efe548901636422a013ce3da7fa22bb9c0f6a3b1

Comment (sameena1415) [reply Features/Lookuptable/README.md:36]: Pushed the changes. Commit code for reference: efe548901636422a013ce3da7fa22bb9c0f6a3b1

**Review by kiran-tripathy — APPROVED** [review]

**Pull request merged**